tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,162 @@
+    $instance->ssh_assert_script_run('sudo systemctl status rke2-server',
+        timeout => 100);
+    record_info('Installed RKE2');
+
+    # Install kubectl
+    $instance->ssh_assert_script_run("curl -LO $kubectl_url",
+        timeout => 1000);
+    $instance->ssh_assert_script_run('chmod +x ./kubectl', timeout => 100);
+    $instance->ssh_assert_script_run('sudo mv ./kubectl /usr/local/bin/kubectl',
+        timeout => 100);
+    $instance->ssh_assert_script_run('kubectl version --client',
+        timeout => 100);
+    record_info('kubectl setup complete');
+
+    # Install Helm
+    $instance->ssh_assert_script_run("curl $helm_url | bash",
+        timeout => 1000);
+    record_info('Helm installation complete');
+}
+
+sub run {
+    my ($self, $args) = @_;
+
+    # Required tools
+    my $ins_rke2 = get_var('RKE2_URL');
+    my $ins_kubectl = get_var('KUBECTL_URL');
+    my $ins_helm = get_var('HELM_URL');
+
+    # Create AWS instance
+    my $provider;
+    my $instance;
+
+    #credentials from openwebui
+    #my $admin_username = get_var('OPENWEBUI_ADMIN');
+    #my $admin_password = get_var('OPENWEBUI_PASSWD');
+    #my $openwebui_hostname = get_var('OPENWEBUI_HOSTNAME');
+
+    select_host_console();
+
+    my $aws_ai_test =
+      get_var('PUBLIC_CLOUD_INSTANCE_FOR_AWS_AI', 'basic_sanity');
+    record_info("DEBUG", "Value of PUBLIC_CLOUD_INSTANCE_FOR_AWS_AI: $aws_ai_test");
+    if ($aws_ai_test) {
+        $instance = $self->{my_instance} = $args->{my_instance};
+        $provider = $self->{provider} =
+          $args->{my_provider};    # Required for cleanup
+        record_info("DEBUG", "my_instance from args: " . (defined($args->{my_instance}) ? $args->{my_instance} : "undefined"));
+        record_info("DEBUG", "my_provider from args: " . (defined($args->{my_provider}) ? $args->{my_provider} : "undefined"));
+    }
+    else {
+        $provider = $self->{provider} =
+          $self->provider_factory();    # Required for cleanup
+        record_info("DEBUG", "Provider created: " . (defined($provider) ? "defined" : "undefined"));
+        $instance = $self->{my_instance} = $provider->create_instance(
+            check_guestregister => is_openstack ? 0 : 1);
+        record_info("DEBUG", "Instance created: " . (defined($instance) ? "defined" : "undefined"));
+    }
+
+    # Debug statements
+    if (defined($instance)) {
+        record_info("DEBUG", "Instance is defined: " . $instance);
+    } else {
+        record_info("DEBUG", "Instance is undefined after assignment.");
+    }
+
+    if (defined($provider)) {
+        record_info("DEBUG", "Provider is defined: " . $provider);
+    } else {
+        record_info("DEBUG", "Provider is undefined after assignment.");
+    }
+
+    if (is_byos()) {
+        registercloudguest($instance);
+        record_info("DEBUG", "Called registercloudguest for BYOS with instance: " . $instance);
+    }
+
+    if (is_byos() && !$aws_ai_test) {
+        registercloudguest($instance);
```

Comment:
this registration already done before: did You mean nested `if` byos, then if Is or not AWS?

Reply:
Fixed.. Taking different approach for registeration